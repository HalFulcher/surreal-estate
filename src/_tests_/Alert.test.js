import React from "react";
import { render } from "@testing-library/react";
import Alert from "../components/Alert";

describe("Alert", () => {
  it("shows error message", () => {
    const { asFragment, getByText } = render(<Alert message="Error!" />);
    expect(getByText(/Error/).textContent).toBe("Error!");
    expect(asFragment()).toMatchSnapshot();
  });

  it("shows success message", () => {
    const { asFragment, getByText } = render(
      <Alert message="Success!" success />
    );
    expect(getByText(/Success/).textContent).toBe("Success!");
    expect(asFragment()).toMatchSnapshot();
  });

  it("doesn't render if message props is empty", () => {
    const { asFragment } = render(<Alert message="" />);
    const alert = asFragment();
    expect(alert).toMatchSnapshot();
  });
});
