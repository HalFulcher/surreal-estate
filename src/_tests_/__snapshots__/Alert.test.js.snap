// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert doesn't render if message props is empty 1`] = `<DocumentFragment />`;

exports[`Alert shows error message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert shows success message 1`] = `
<DocumentFragment>
  <div
    class="Alert alert-success"
  >
    Success!
  </div>
</DocumentFragment>
`;
